src/client/renderingengine.cpp

```diff
@@ -148,6 +151,10 @@ RenderingEngine::RenderingEngine(IEventReceiver *receiver)
 			+ "shaders" + DIR_DELIM + "Irrlicht";
 	params.OGLES2ShaderPath = (porting::path_share + DIR_DELIM + rel_path + DIR_DELIM).c_str();
 
+#if BUILD_HEADLESS
+	SDL_VideoInit("offscreen");
```

Comment:
You checked this doesn't break anything if BUILD_HEADLESS is true but the client is not set for headless?

Reply:
will check tmrw in office

Reply:
No it doesn't work anymore. If you do BUILD_HEADLESS, you cannot run non-headless. If you don't BUILD_HEADLESS, you cannot run headless. This is a pretty bad interface, we should probably get rid of the CLI flag completely and just have it be set during build?

Reply:
Even better, the Python client should check if the binary was actually built with headless enabled.

Reply:
I attempted to fix it so that it can render to the screen even if built with headless mode, though I haven't tested it.